// react-envelope/lib/withState.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.default = withState;

var _getEnhancedComponent = require('./utils/getEnhancedComponent');

var _getEnhancedComponent2 = _interopRequireDefault(_getEnhancedComponent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function withState(definition) {
  return function (BaseComponent) {
    var EnhancedComponent = (0, _getEnhancedComponent2.default)(BaseComponent);

    function prepareState() {
      var _this = this;

      var initialState = {};
      var stateMutators = {};

      Object.keys(definition).forEach(function (key) {
        var setState = function setState(input) {
          _this.setState(function (prevState) {
            var value = typeof input === 'function' ? input(prevState[key]) : input;

            return _extends(_defineProperty({}, key, value), prevState[key]);
          });
        };

        initialState[key] = definition[key].initial;
        stateMutators[key] = definition[key].mutators(setState);
      });

      this.state = initialState;
      this.stateMutators = stateMutators;
    }

    function createAdvancedState(state) {
      var _this2 = this;

      if (!state) return undefined;

      var advancedState = {};

      Object.keys(state).forEach(function (key) {
        advancedState[key] = _extends({}, _this2.state[key], _this2.stateMutators[key]);
      });

      return advancedState;
    }

    // prototype
    EnhancedComponent.prototype.prepareState = prepareState;
    EnhancedComponent.prototype.createAdvancedState = createAdvancedState;

    return EnhancedComponent;
  };
}
